### ISSUE TITLE
Style ProgressIterator without overwriting length 

### ISSUE THREAD

**Issue by LeCyberDucky** [mitsuhiko/indicatif#305]
Hi, 

I'm trying to use a `ProgressIterator` by following the example given in the [documentation](https://docs.rs/indicatif/0.16.2/indicatif/#iterators): 

```
use indicatif::ProgressIterator;

for _ in (0..1000).progress() {
    // ...
}
```

I would like to style my progress bar, however. A bit further down in the documentation, the following snippet is given: 

```
use indicatif::{ProgressBar, ParallelProgressIterator};
use rayon::iter::{ParallelIterator, IntoParallelRefIterator};

// Use `ProgressBar::with_style` to change the view
let pb = ProgressBar::new();
let v: Vec<_> = (0..100000).collect();
let v2: Vec<_> = v.par_iter().progress_with(pb).map(|i| i + 1).collect();
assert_eq!(v2[0], 1);
```

Unlike the example, however, I cannot call `ProgressBar::new()` without specifying a length. Is this limited to these parallel iterators? I only include `indicatif::{ProgressBar, ProgressIterator, ProgressStyle}`. 

So, basically, I just want to provide my `ProgressIterator` with a `ProgressStyle`, but it appears that I need to do that in the form of a new `ProgressBar`. That would be fine, if I could do that without overwriting the length that is automatically determined by the `ProgressIterator`. Is that possible?

**Comment by djc** [comment]
Yeah, it looks like we're not testing that example, and I guess what you want is not currently possible. Why don't you call `iter.len()` and pass that into the new `ProgressBar` before wrapping the iterator in the `ProgressIterator`?

**Comment by LeCyberDucky** [comment]
Alright, thank you. Yeah, I've resorted to a work-around similar to what you suggest. From the documentation I just thought that it *should* work and that I was simply missing something. I still think that it would be nice to implement a solution for this, so I'll leave the issue open, but feel free to close it, if you don't think it should be left open.

**Comment by djc** [comment]
If you like, I'd be open to reviewing a change that adds `ProgressIterator::progress_with_style()` that takes a `ProgressStyle` as an argument but otherwise uses the same logic as `ProgressIterator::progress()`.

**Comment by LeCyberDucky** [comment]
I'll give that a try 👍
